src/puppetlabs/puppetdb/catalogs.clj

```diff
@@ -158,7 +158,9 @@
    :file (s/maybe s/Str)
    :line (s/maybe s/Int)
    :parameters {s/Any s/Any}
-   :resource s/Str
+   ;; On historical-catalogs this is an optional-key but this is required on
+   ;; "normal" catalogs
+   (s/optional-key :resource) s/Str
```

Comment:
I'd rather this be addressed by us storing resource hashes. I think you could find uses for them and I think not having them introduces unneeded inconsistency between the historical and present day views.